src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -36,6 +37,20 @@ public boolean contains(Person toCheck) {
         return internalList.stream().anyMatch(toCheck::isSamePerson);
     }
 
+
+    /**
+     * Returns student in the list that matches the name {@code toCheck}.
+     */
+    public Person matchName(Name toCheck) {
+        requireNonNull(toCheck);
+        Optional<Person> matchingStudent = internalList.stream().filter(p -> p.isSameName(toCheck)).findAny();
+        if (matchingStudent.isEmpty()) {
```

Comment:
I believe you can simplify the next few lines into a single line using `orElseThrow(Supplier<? extends X> exceptionSupplier)` cause we're using Optionals. But not that big of a deal